scanner/src/main/kotlin/Scanner.kt

```diff
@@ -83,6 +84,58 @@ class Scanner(
     private val archiver: FileArchiver? = scannerConfig.archive.createFileArchiver(),
     fileListStorage: ProvenanceFileStorage = scannerConfig.fileListStorage.createStorage()
 ) {
+    companion object {
```

Comment:
Up for discussion: Should we make these top-level functions instead to reduce nesting? I'm not sure what the value for is for them to be scoped under `Scanner`.